apps/framework-cli-e2e/test/cli.test.ts

```
@@ -0,0 +1,212 @@
+      stdio: "pipe",
+      cwd: TEST_PROJECT_DIR,
+    });
+
+    await new Promise<void>((resolve, reject) => {
+      let serverStarted = false;
+      devProcess!.stdout?.on("data", async (data) => {
+        const output = data.toString();
+        console.log("Dev server output:", output);
+
+        if (
+          !serverStarted &&
+          output.includes(
+            "Your local development server is running at: http://localhost:4000/ingest",
+          )
+        ) {
+          resolve();
+          serverStarted = true;
+        }
+      });
+
+      devProcess!.stderr?.on("data", (data) => {
+        console.error("Dev server stderr:", data.toString());
+      });
+
+      devProcess!.on("exit", (code) => {
+        console.log(`Dev process exited with code ${code}`);
+        expect(code).to.equal(0);
+      });
+
+      (async () => {
+        await setTimeoutAsync(60_000);
+        if (devProcess!.killed) return;
+        console.error("Dev server did not start or complete in time");
+        devProcess!.kill("SIGINT");
+        reject(new Error("Dev server timeout"));
+      })();
+    });
+
+    console.log("Server started, waiting before sending test request...");
+
+    await setTimeoutAsync(2000);
+
+    console.log("Sending test request...");
+    const eventId = randomUUID();
+    try {
+      const response = await fetch(
+        "http://localhost:4000/ingest/UserActivity",
+        {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            eventId: eventId,
+            timestamp: "2019-01-01 00:00:01",
+            userId: "123456",
+            activity: "click",
+          }),
+        },
+      );
+
+      if (response.ok) {
+        console.log("Test request sent successfully");
+      } else {
+        console.error("Response code:", response.status);
+        console.error(`Test request failed:`, await response.text());
```

Should we fail early and exit here?